packages/aztec-cli/scripts/bot.js

```diff
@@ -1,164 +1,179 @@
+			const tokens = data.msg.split(' ');
+			const payeeAddress = tokens[1];
+			const balance = new BN(await c.eth.getBalance(payeeAddress));
+			const result = await topUpFromMessage({
+				balance,
+				payeeAddress,
+			});
+			const { txHash } = result.toJS();
+			c.constructAndSendMessage(
+				'msg',
+				JSON.stringify({
+					txNoteHash,
+					requestNum,
+					timestamp: Date.now(),
+				}).toString(),
+				data['fromPublicKey']
+			);
+		}
+	}
+});
+
+const mintFromMessage = async ({ minteeAddress, minteePublicKey, mintFromZero = false, mintAmount, tradeSymbol }) => {
+	console.log(`Address   : ${minteeAddress}`);
+	console.log(`Public Key: ${minteePublicKey}`);
+
+	const config = getConfig();
+	const deployerAddress = config['DEPLOYER_ADDRESS'];
+	const deployerPassword = config['DEPLOYER_PASSWORD'];
+
+	const result = await mint(
+		Map({
+			tradeSymbol: tradeSymbol || 'AAA',
+			minteeAddress: minteeAddress,
+			minteePublicKey: minteePublicKey,
+			deployerAddress: deployerAddress,
+			deployerPassword: deployerPassword,
+			minteeAmount: mintAmount || new BN(0),
+			mintFromZero: mintFromZero,
+		})
+	);
+
+	console.log('minteeNoteHash', result.get('minteeNoteHash'));
+	console.log('Minting complete.');
+	wallet.shutdownSync();
+	return result;
+};
+
+const PAY_AMOUNT = new BN(toWei('0.1', 'ether'));
+
+const topUpFromMessage = async ({ payeeAddress, balance }) => {
+	console.log(`Address   : ${minteeAddress}`);
+	console.log(`Public Key: ${minteePublicKey}`);
+
+	const config = getConfig();
+	const deployerAddress = config['DEPLOYER_ADDRESS'];
+	const deployerPassword = config['DEPLOYER_PASSWORD'];
+
+	if (balance.lt(limit)) {
+		const { signerEth } = await wallet.prepareSigner({
+			deployerAddress,
+			deployerPassword,
+		});
+
+		await wallet.init();
+
+		const result = await wallet.pay({
+			minteeAddress: minteeAddress,
+			minteePublicKey: minteePublicKey,
+			fromAddress: deployerAddress,
+			toAddress: payeeAddress,
+			payAmount: PAY_AMOUNT,
+			deployerPassword: deployerPassword,
+			minteeAmount: mintAmount || new BN(0),
+			mintFromZero: mintFromZero,
+		});
+
+		console.log('minteeNoteHash', result.get('minteeNoteHash'));
+		console.log('Minting complete.');
+		wallet.shutdownSync();
+		return result;
+	} else {
+		return Map({});
+	}
+};
 
 const main = async () => {
-    await c.start()
-    await new Promise((resolve, reject) => {
-      setTimeout(async () => {
-        await c.client.write(JSON.stringify({
-          type: 'join', cname: 'zktransfer', uname: 'darkbot'
-        }))
-        await c.client.write(JSON.stringify({
-          type: 'msg', uname: 'darkbot', msg: 'DARKBOT LIVES'
-        }).toString())
-        resolve()
-      }, 1000)
-    })
-}
-
-main().then(() => console.log("Exited"))
+	await c.start();
+	await new Promise((resolve, reject) => {
+		setTimeout(async () => {
+			c.uname = 'darkbot';
+			c.channel = 'zktransfer';
+			await c.constructAndSendMessage('join', '');
```

Comment:
This is a good place to start. We'll want to follow the example in `distributed_chat/tests/dm.spec.js` for connecting (requesting a challenge) and returning the signature of that challenge.

I believe just trying to join a public channel here will cause a server error before we've registered our public key.

Reply:
The public key is generated in the start() function.